web-app/src/common/pathfinding.ts

```diff
@@ -28,3 +28,24 @@ export function getWinningPathIfExist(
   }
   return { hasWon: false, winningPath: null };
 }
+
+export function getWinnablePathCost(
+  gameState: GameState,
+  stoneColor: StoneColor,
+): number {
+  const hexBoardWinPredictionGraph = createWinPredictionGraph(gameState, stoneColor);
+  const dijkstra = new jkstra.algos.Dijkstra(hexBoardWinPredictionGraph.winPrediction);
+  let shortestPath = dijkstra.shortestPath(
+    hexBoardWinPredictionGraph.vertexMap.get(stoneColor === "black" ? BLACK_NODE_START : WHITE_NODE_START),
+    hexBoardWinPredictionGraph.vertexMap.get(stoneColor === "black" ? BLACK_NODE_END : WHITE_NODE_END),
+    {
+      edgeCost: function (edge) {
+        return edge.data;
+      }
+    });
+  if (!shortestPath) return -1;
+  const cost = Object.keys(shortestPath).reduce(function (previous, key) {
+    return shortestPath[key].data === UNDEFINED_EDGE_COST ? previous : previous + shortestPath[key].data
+  }, 0);
+  return cost;
+}
```

Comment:
Can't this function already return the number of moves needed to win, instead of this 'cost' which, to me, is more of a computation intermediate than a real result?

Reply:
Yes !